ttg/ttg/terminal.h

```diff
@@ -258,6 +248,21 @@ namespace ttg {
     using input_terminals_tuple_t = typename input_terminals_tuple<keyT, valuesT...>::type;
   }  // namespace detail
 
+  namespace meta {
+    /// detects whether a given type is an input terminal type
+    template <typename T>
+    inline constexpr bool is_input_terminal_v = false;
+    template <typename keyT>
+    inline constexpr bool is_input_terminal_v<InTerminalBase<keyT>> = true;
+    template <typename keyT, typename valueT>
+    inline constexpr bool is_input_terminal_v<In<keyT, valueT>> = true;
+    template <>
+    inline constexpr bool is_input_terminal_v<TerminalBase> = true;
```

Comment:
A `TerminalBase` is always an input terminal?

Reply:
fixed